lib/GenericsAPI/Utils/MatrixUtil.py

```diff
@@ -177,23 +177,37 @@ def _process_attribute_mapping_sheet(self, file_path, sheet_name, matrix_name, w
 
             return ref.get('attribute_mapping_ref')
 
-    def _file_to_data(self, file_path, refs, matrix_name, workspace_id):
-        log('Start reading and converting excel file data')
-        data = refs
+    def _file_to_df(self, file_path):
+        log('start parsing file content to data frame')
 
         try:
-            df = pd.read_excel(file_path, sheet_name='data')
+            df = pd.read_excel(file_path, sheet_name='data', index_col=0)
 
         except XLRDError:
             try:
-                df = pd.read_excel(file_path)
+                df = pd.read_excel(file_path, index_col=0)
                 print('WARNING: A sheet named "data" was not found in the attached file, '
                       'proceeding with the first sheet as the data sheet.')
 
             except XLRDError:
-                df = pd.read_csv(file_path, index_col=0)
+
+                try:
+                    reader = pd.read_csv(file_path, sep=None, iterator=True)
+                    inferred_sep = reader._engine.data.dialect.delimiter
+                    df = pd.read_csv(file_path, sep=inferred_sep, index_col=0)
+                except Exception:
+                    raise ValueError('Cannot parse file. Please provide valide excel or csv file')
```

Comment:
Maybe "Please provide valid excel, CSV or TSV file"

Do you think we should print or check the `inferred_sep`? If not, you can just use `sep=None` in line 197